### PR TITLE
Add FXIOS-7562 [v121] addressAutofill on/off in nimbus

### PR DESCRIPTION
**Pull request by jnrahme**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7562)

## :bulb: Description
Adding a feature flag for address autofill in Nimbus 

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by jnrahme** [comment]
Screenshot 2023-10-23 at 12 34 55 PM
Screenshot 2023-10-23 at 12 39 51 PM
Tested Feature flag with print statements in viewDidLoad in BrowserViewController

File: Client/Nimbus/NimbusFeatureFlagLayer.swift [hunk 1]
```diff
@@ -63,17 +69,14 @@ final class NimbusFeatureFlagLayer {
 
         case .tabTrayRefactor:
             return checkTabTrayRefactorFeature(from: nimbus)
-
+
```
Comment (adudenamedruby) [inline Client/Nimbus/NimbusFeatureFlagLayer.swift:72]: whitespace

File: nimbus-features/addressAutofillFeature.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# The configuration for the addressAutofillFeature feature
+features:
+  address-autofill-feature:
+    description: >
+      Feature description
```
Comment (adudenamedruby) [inline nimbus-features/addressAutofillFeature.yaml:5]: add a description. lolol

File: nimbus-features/addressAutofillFeature.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    variables:
+      status:
+        description: >
+          Variable description
```
Comment (adudenamedruby) [inline nimbus-features/addressAutofillFeature.yaml:9]: add a description lololol

**Review by adudenamedruby — CHANGES_REQUESTED** [review]

**Review by adudenamedruby — APPROVED** [review]

**Review by yoanarios — APPROVED** [review]

File: .experimenter.yaml [hunk 4]
```diff
@@ -1,4 +1,12 @@
 ---
+address-autofill-feature:
+  description: "Feature description\n"
```
Comment (lmarceau) [inline .experimenter.yaml:3]: @adudenamedruby is this normal this is `"Feature description\n"`only? Here and under the variables.description as well

Comment (jnrahme) [reply .experimenter.yaml]: Good catch I pushed a description

**Pull request merged**